blaze/compute/sql.py

```diff
@@ -262,19 +262,41 @@ def compute_up(t, s, **kwargs):
 @dispatch(Selection, sa.sql.ColumnElement)
 def compute_up(expr, data, scope=None, **kwargs):
     predicate = compute(expr.predicate, data, post_compute=False)
-    return sa.select([data]).where(predicate)
+    return compute_up(expr, data, predicate, scope=scope, **kwargs)
+
+
+@dispatch(Selection, sa.sql.ColumnElement, ClauseElement)
```

Comment:
Is `BinaryExpression` too restrictive here? Are there any valid predicate functions that don't fall into this? Is there any `ClauseElement` that we would not treat as a valid predicate?